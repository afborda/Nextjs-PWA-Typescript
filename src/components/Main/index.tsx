const Main = () => (
  <main>
    <h2>Abner fonseca</h2>
  </main>
)

export default Main
