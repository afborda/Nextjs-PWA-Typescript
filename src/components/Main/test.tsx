import { screen, render } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should validate text Abner fonseca', () => {
    render(<Main />)

    expect(
      screen.getByRole('heading', { name: /Abner fonseca/i })
    ).toBeInTheDocument()
  })

  it('should validate Subtitle text ', () => {
    render(<Main />)

    expect(
      screen.getByRole('heading', {
        name: /Poc de Typescript, styled-components, Reactjs, NextJs/i
      })
    ).toBeInTheDocument()
  })
  it('should  render the color correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({
      'background-color': '#06092b'
    })
  })

  it('should render correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
