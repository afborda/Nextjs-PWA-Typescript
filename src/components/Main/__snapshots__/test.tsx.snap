// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render correctly 1`] = `
<main
  class="styles__Wrapper-sc-1qrkey8-0 jeLfuH"
>
  <img
    alt="imagem de um àtomo do react avançado"
    class="styles__Logo-sc-1qrkey8-3 fawfXS"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-sc-1qrkey8-1 eXCKpt"
  >
    Abner Fonseca
  </h1>
  <h2
    class="styles__Subtitle-sc-1qrkey8-2 fdqClS"
  >
    Poc de Typescript, styled-components, Reactjs, NextJs
  </h2>
  <img
    alt="um desenvolvedor olhando para tela do computador"
    class="styles__Ilustrator-sc-1qrkey8-4 gilZoB"
    src="/img/illustration.svg"
  />
</main>
`
